src/coreclr/vm/eetwain.cpp

```diff
@@ -4712,16 +4714,41 @@ bool EECodeManager::EnumGcRefs( PREGDISPLAY     pContext,
             {
                 _ASSERTE(willContinueExecution);
 
+#ifdef FEATURE_EH_FUNCLETS
+                // Funclets' frame pointers(EBP) are always restored so they can access to main function's local variables.
+                // Therefore the value of EBP is invalid for unwinder so we should use ESP instead.
+                // See UnwindStackFrame for details.
+                if (pCodeInfo->IsFunclet())
+                {
+                    PTR_CBYTE methodStart = PTR_CBYTE(pCodeInfo->GetSavedMethodCode());
+                    TADDR baseSP = ESP;
+                    // Set baseSP as initial SP
+                    baseSP += GetPushedArgSize(&info, table, curOffs);
+                    // 16-byte stack alignment padding (allocated in genFuncletProlog)
+                    // Current funclet frame layout (see CodeGen::genFuncletProlog() and genFuncletEpilog()):
+                    //   prolog: sub esp, 12
+                    //   epilog: add esp, 12
+                    //           ret
+                    // SP alignment padding should be added for all instructions except the first one and the last one.
+                    // Epilog may not exist (unreachable), so we need to check the instruction code.
+                    if (curOffs != 0 && methodStart[curOffs] != X86_INSTR_RETN)
```

Comment:
This condition is actually wrong... It needs to be the same one as in `UnwindStackFrame`...